codec-http/src/test/java/io/netty/handler/codec/http/HttpHeadersTest.java

```diff
@@ -76,8 +76,8 @@ public void testAddSelf() {
         headers.add(headers);
     }
 
-    @Test(expected = IllegalArgumentException.class)
-    public void testSetSelf() {
+    @Test
```

Comment:
So looking at the 4.0 HttpHeader class I think whats in this PR is more compatible with that than what was there before.

If 4.0  headers.set(headers) would not throw but it was semantically broken as it would do

this.clear()
<add all entries>

which would have the unintended consequence of emptying the headers instead of leaving them unchanged. In short there's a nasty bug in 4.0 rather than a semantic incompatability.

I'm happy to create a separate PR to fix the issue in 4.0.

There is no setAll in 4.0